pkg/middleware/cookie_refresh.go

```diff
@@ -0,0 +1,60 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/justinas/alice"
+	middlewareapi "github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/middleware"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/logger"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/requests"
+)
+
+type CookieRefreshOptions struct {
+	IssuerURL string
+}
+
+func NewCookieRefresh(opts *CookieRefreshOptions) alice.Constructor {
+	cr := &cookieRefresh{
+		HttpClient: &http.Client{},
+		IssuerURL:  opts.IssuerURL,
+	}
+	return cr.refreshCookie
+}
+
+type cookieRefresh struct {
+	HttpClient *http.Client
+	IssuerURL  string
+}
+
+func (cr *cookieRefresh) refreshCookie(next http.Handler) http.Handler {
```

Comment:
For clarification, the method name refreshCookie indicates to me that it'll refresh the current session based on the cookie, and the caller adds it to chain calls, but what does that mean when the user should be automatically logged out? For example, if the user remains idle for 'X' minutes (no API interaction to refresh the cookie) it should be automatically redirected to the IAM screen when interacting again, but in this case, the cookie will 'OK' because some other application is keeping it alive

Reply:
I think the cookie is just an identifier for the session, if the IAM session is terminated the cookie is no longer valid even if we try to keep it fresh...